test/shared/src/main/scala/zio/test/Spec.scala

```diff
@@ -312,10 +314,23 @@ final case class Spec[-R, +E, +L, +T](caseValue: SpecCase[R, E, L, T, Spec[R, E,
         val (z, caseValue) = f(z0, t)
         ZIO.succeed(z -> Spec(caseValue))
     }
+
+  /**
+   * Runs only tests whose labels (which must be strings) contain the given substring
+   */
+  final def only[S, E0](
+    s: String
+  )(implicit ev1: L <:< String, ev2: T <:< TestSuccess[S], ev3: E <:< TestFailure[E0]): ZSpec[R, E0, String, S] =
+    mapLabel(ev1)
```

Comment:
If we have the implicit evidence we can use a cast to avoid doing multiple traversals of the `Spec`. `Spec.asInstanceOf[ZSpec[R, E1, String, S].filterLabels(_.contains(s)).getOrElse(Spec.test("only", ignored))`.